src/java.base/share/classes/sun/security/ssl/HandshakeContext.java

```diff
@@ -454,7 +454,8 @@ void dispatch(byte handshakeType,
         if (handshakeType == SSLHandshake.HELLO_REQUEST.id) {
             // For TLS 1.2 and prior versions, the HelloRequest message MAY
             // be sent by the server at any time.
-            consumer = SSLHandshake.HELLO_REQUEST;
+            consumer = conContext.sslConfig.isClientMode ?
```

Comment:
This seems reasonable, but could you update the bug report to say why this fixes the problem?

Reply:
This would an good comment to have in the code.